### DISCUSSION TITLE
Ultrawrap

### DISCUSSION THREAD

**Discussion by justin2004** [INCATools/semantic-sql#18] [General]
Just wanted to share this paper since the idea seems similar to semantic-sql:
https://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.259.2084&rep=rep1&type=pdf
"Ultrawrap: SPARQL Execution on Relational Data Technical Report"

"... comprises  encoding  a  logical  representation  of  the database as a graph using SQL views and a simple syntactic translation of SPARQL queries to SQL queries on those views"

I think Ultrawrap lead to https://gra.fo/.

**Comment by jamesaoverton** [comment]
Thanks, I hadn't heard of Ultrawrap. There's a number of SPARQL to SQL implementations out there, but I'm not aware of one that is (1) open, (2) maintained, and (3) performant. Does anybody have a good suggestion?

It should be possible to use the [Linked Data Fragments](https://linkeddatafragments.org/) approach with RDFTab: The LDF client breaks a SPARQL query down into a number of quad/triple queries, and it would be easy to serve these quads/triples from an RDFTab table. I haven't tried this yet. The overall approach deliberately sacrifices some query performance to reduce the computational load on the server.

[R2RML](https://www.w3.org/TR/r2rml/) is related, but I've had mixed experiences trying to use it. I just found [RML](https://rml.io/) which looks promising.

**Reply by justin2004** [reply]
there is https://ontop-vkg.org/ which uses R2RML mappings to do query answering (SPARQL to SQL).

RML looked promising to me too a few months ago until i tried to use it (with json sources). it has a number of limitations around jsonpath which made it not suitable for my team's needs. i noted some of those limitations and my alternative solution here: https://github.com/justin2004/rml-testing

**Reply by justin2004** [reply]
and here are some more SPARQL to SQL related tools:
https://github.com/semantalytics/awesome-semantic-web#rdb-to-rdf
